cf-agent/cf-agent.c

```diff
@@ -308,6 +310,11 @@ static GenericAgentConfig *CheckOpts(int argc, char **argv)
             TIMING = true;
             break;
 
+        case 'w':
+            Log(LOG_LEVEL_INFO, "Setting workdir to '%s'", optarg);
+            putenv("CFENGINE_TEST_OVERRIDE_WORKDIR", xstrdup(optarg), 1);
```

Comment:
`putenv` takes only one argument, a string of the form `CFENGINE_TEST_OVERRIDE_WORKDIR=/var/cfengine`, and this complete string is what you need to `strdup`. If `CFENGINE_TEST_OVERRIDE_WORKDIR` exists in the environment already, it will replace the existing string.